src/fs.register.ts

```diff
@@ -31,8 +32,40 @@ export const fqdn: FinalizeRequestMiddleware<object, MetadataBearer> = (next) =>
   };
 };
 
+/**
+ * AWS SDK middleware logic to try 3 times if receiving an EAI_AGAIN error
+ */
+export function eaiAgainBuilder(timeout: (attempt: number) => number): BuildMiddleware<object, MetadataBearer> {
+  const eaiAgain: BuildMiddleware<object, MetadataBearer> = (next) => {
+    const maxTries = 3;
+    return async (args) => {
+      for (let attempt = 1; attempt <= maxTries; attempt++) {
+        try {
+          return await next(args);
+        } catch (error) {
+          if (error && typeof error === 'object' && 'code' in error && 'hostname' in error) {
+            if (error.code !== 'EAI_AGAIN') {
+              throw error;
+            }
+            logger.warn({ host: error.hostname }, `eai_again retry #${attempt}`);
```

Comment:
its best to structure log the attempt counter, of data so it can be filtered on later.

Could we also add totalDelay? and log the next dealy? 

```suggestion
            const delay = timeout(attempt);
            logger.warn({ host: error.hostname, attempt, delay, totalDelay }, `eai_again:retry`);
            totalDelay += delay;
```